pkg/phlaredb/symdb/resolver_stacktraces.go

```diff
@@ -0,0 +1,171 @@
+package symdb
+
+import (
+	"github.com/parquet-go/parquet-go"
+
+	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
+	schemav1 "github.com/grafana/pyroscope/pkg/phlaredb/schemas/v1"
+)
+
+// CallSiteValues represents statistics associated with a call tree node.
+type CallSiteValues struct {
+	// Flat is the sum of sample values directly attributed to the node.
+	Flat uint64
+	// Total is the total sum of sample values attributed to the node and
+	// its descendants.
+	Total uint64
+	// LocationFlat is the sum of sample values directly attributed to the
+	// node location, irrespectively of the call chain.
+	LocationFlat uint64
+	// LocationTotal is the total sum of sample values attributed to the
+	// node location and its descendants, irrespectively of the call chain.
+	LocationTotal uint64
+}
```

Comment:
Please note that at the moment only `Total` is actually used. We can return all, but we need some toggle in UI that switches between them. Otherwise, the timeline view becomes too busy (especially, if `group by` is used)